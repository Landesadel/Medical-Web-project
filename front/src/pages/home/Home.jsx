import React from 'react';
import TopSection from "../../components/TopSection/TopSection";
import Courses from "../../components/Courses/Courses";
import News from "../../components/News/News";
// import Courses from "../../Components/Courses/Courses";
// import News from "../../Components/News/News";
// import Sources from "../../Components/Sources/Sources";


const Home = () => {
    return (
        <div>
           <TopSection/>
            <Courses/>
            <News/>
            {/*<Sources/>*/}
        </div>
    );
};

export default Home;