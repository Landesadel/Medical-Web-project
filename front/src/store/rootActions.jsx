import * as userActions from './user/user.actions';
import * as forumActions from './forum/forum.actions';

export const allActions = {
	...userActions,
	...forumActions,
};
