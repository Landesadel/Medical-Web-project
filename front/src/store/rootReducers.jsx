import { reducer as userReducer } from './user/user.slice';
import { reducer as forumReducer } from './forum/forum.slice';
import { reducer as toastrReducer } from 'react-redux-toastr';

export const reducers = {
	user: userReducer,
	toastr: toastrReducer,
	forum: forumReducer,
};
