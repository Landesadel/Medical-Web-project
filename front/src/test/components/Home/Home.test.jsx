import React from 'react';
import {render, screen} from '@testing-library/react';
import '@testing-library/jest-dom';
import {BrowserRouter, Routes} from 'react-router-dom';
import Home from "../../../pages/home/Home";
import {QueryClient, QueryClientProvider} from "react-query";
import {store} from "../../../store/store";
import {Provider} from "react-redux";
import {MemoryRouter} from "react-router-dom";


describe('Home', () => {

    const queryClient = new QueryClient();
    it('renders without crashing', () => {
        const snapshot = render(
            <Provider store={store}>
                <MemoryRouter>
                    <QueryClientProvider client={queryClient}>
                        <Home/>
                    </QueryClientProvider>
                </MemoryRouter>
            </Provider>
        );
        expect(screen.getByTestId('top-section')).toBeInTheDocument();
        expect(screen.getByTestId('news')).toBeInTheDocument();
        expect(screen.getByTestId('centers-slider')).toBeInTheDocument();
        expect(screen.getByTestId('conf-and-seminars')).toBeInTheDocument();
        expect(screen.getByTestId('sources')).toBeInTheDocument();
        expect(snapshot).toMatchSnapshot();
    });

});
