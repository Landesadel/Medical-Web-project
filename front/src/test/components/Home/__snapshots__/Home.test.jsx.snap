// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="topSection"
          data-testid="top-section"
          id="topSection"
        >
          <div
            class="container"
          >
            <div
              class="row d-flex align-items-center"
            >
              <div
                class="col-sm"
              >
                <div
                  class="topWrapper"
                >
                  <h1
                    class="topHeading"
                  >
                    Белорусское общество симуляционного обучения в медицине
                  </h1>
                  <p
                    class="topDescription"
                  >
                    БОСОМ объединяет профессионалов в области различных направлений медицины и ставит целью широкое внедрение в медицинскую практику симуляционных технологий в медицинском образовании, повышение квалификации медицинских работников, проведение сертификации и аттестации, внедрение передовых технологий в сфере здравоохранения
                  </p>
                  <a
                    aria-current="page"
                    class="button"
                    href="/history"
                  >
                    Подробнее
                  </a>
                </div>
              </div>
              <div
                class="col-lg p-0 topImage"
              >
                <img
                  alt="image"
                  class="photoSlider"
                  src=""
                />
                <img
                  alt="image"
                  class="photoSlider"
                  src=""
                />
                <img
                  alt="image"
                  class="photoSlider"
                  src=""
                />
                <img
                  alt="image"
                  class="photoSlider"
                  src=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="news"
          data-testid="news"
          id="news"
        >
          <div
            class="container"
          >
            <div
              class="row"
            >
              <div
                class="col"
              >
                <h2
                  class="newsHeading"
                >
                  Новости и события
                </h2>
              </div>
            </div>
            <div
              class="row"
            >
              <h4>
                Loading...
              </h4>
            </div>
            <div
              class="row"
            >
              <div
                class="col"
              >
                <a
                  class="allEvents"
                  href="/news"
                >
                  Все новости
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          class="courses"
          data-testid="centers-slider"
          id="courses"
        >
          <div
            class="container"
          >
            <div
              class="row"
            >
              <div
                class="col"
              >
                <h2
                  class="coursesHeading"
                >
                  Симуляционные центры
                </h2>
              </div>
            </div>
            <div
              class="row"
            >
              <div
                class="col"
              >
                <div
                  class="sliderWrapper"
                >
                  <div>
                    <div
                      class="slick-slider slick-initialized"
                    >
                      <div
                        class="slick-list"
                        style="height: 0px;"
                      >
                        <div
                          class="slick-track"
                          style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="row"
            >
              <div
                class="col"
              >
                <a
                  class="allEvents"
                  href="/centers"
                >
                  Смотреть все центры
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          class="news"
          data-testid="conf-and-seminars"
          id="news"
        >
          <div
            class="container"
          >
            <div
              class="row"
            >
              <div
                class="col"
              >
                <h2
                  class="newsHeading"
                >
                  Конференции и семинары
                </h2>
              </div>
            </div>
            <div
              class="row"
            >
              <h4>
                Loading...
              </h4>
            </div>
            <div
              class="row"
            >
              <div
                class="col"
              >
                <a
                  class="allEvents"
                  href="/conferencies"
                >
                  Смотреть все события
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          class="sources"
          data-testid="sources"
          id="sources"
        >
          <div
            class="container"
          >
            <div
              class="row"
            >
              <div
                class="col"
              >
                <h2
                  class="sourcesHeading"
                >
                  Обучающие материалы
                </h2>
              </div>
            </div>
            <div
              class="row"
            >
              <div
                class="sources"
              >
                <div
                  class="me-auto mb-5 sourcesItem"
                >
                  <div
                    class="sourcesIcon"
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-book "
                      data-icon="book"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      viewBox="0 0 448 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M96 0C43 0 0 43 0 96V416c0 53 43 96 96 96H384h32c17.7 0 32-14.3 32-32s-14.3-32-32-32V384c17.7 0 32-14.3 32-32V32c0-17.7-14.3-32-32-32H384 96zm0 384H352v64H96c-17.7 0-32-14.3-32-32s14.3-32 32-32zm32-240c0-8.8 7.2-16 16-16H336c8.8 0 16 7.2 16 16s-7.2 16-16 16H144c-8.8 0-16-7.2-16-16zm16 48H336c8.8 0 16 7.2 16 16s-7.2 16-16 16H144c-8.8 0-16-7.2-16-16s7.2-16 16-16z"
                        fill="currentColor"
                      />
                    </svg>
                  </div>
                  <h3
                    class="sourcesTitle"
                  >
                    Литература
                  </h3>
                  <p
                    class="sourcesText"
                  >
                    Методические пособия, книги, презентации
                  </p>
                  <a
                    class="link"
                    href="/login"
                  >
                    Подробнее...
                  </a>
                </div>
                <div
                  class="me-auto mb-5 sourcesItem"
                >
                  <div
                    class="sourcesIcon"
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-video "
                      data-icon="video"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      viewBox="0 0 576 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M0 128C0 92.7 28.7 64 64 64H320c35.3 0 64 28.7 64 64V384c0 35.3-28.7 64-64 64H64c-35.3 0-64-28.7-64-64V128zM559.1 99.8c10.4 5.6 16.9 16.4 16.9 28.2V384c0 11.8-6.5 22.6-16.9 28.2s-23 5-32.9-1.6l-96-64L416 337.1V320 192 174.9l14.2-9.5 96-64c9.8-6.5 22.4-7.2 32.9-1.6z"
                        fill="currentColor"
                      />
                    </svg>
                  </div>
                  <h3
                    class="sourcesTitle"
                  >
                    Видеоматериалы
                  </h3>
                  <p
                    class="sourcesText"
                  >
                    Обучающие видеоролики, конференции, лекции
                  </p>
                  <a
                    class="link"
                    href="/login"
                  >
                    Подробнее...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="topSection"
        data-testid="top-section"
        id="topSection"
      >
        <div
          class="container"
        >
          <div
            class="row d-flex align-items-center"
          >
            <div
              class="col-sm"
            >
              <div
                class="topWrapper"
              >
                <h1
                  class="topHeading"
                >
                  Белорусское общество симуляционного обучения в медицине
                </h1>
                <p
                  class="topDescription"
                >
                  БОСОМ объединяет профессионалов в области различных направлений медицины и ставит целью широкое внедрение в медицинскую практику симуляционных технологий в медицинском образовании, повышение квалификации медицинских работников, проведение сертификации и аттестации, внедрение передовых технологий в сфере здравоохранения
                </p>
                <a
                  aria-current="page"
                  class="button"
                  href="/history"
                >
                  Подробнее
                </a>
              </div>
            </div>
            <div
              class="col-lg p-0 topImage"
            >
              <img
                alt="image"
                class="photoSlider"
                src=""
              />
              <img
                alt="image"
                class="photoSlider"
                src=""
              />
              <img
                alt="image"
                class="photoSlider"
                src=""
              />
              <img
                alt="image"
                class="photoSlider"
                src=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="news"
        data-testid="news"
        id="news"
      >
        <div
          class="container"
        >
          <div
            class="row"
          >
            <div
              class="col"
            >
              <h2
                class="newsHeading"
              >
                Новости и события
              </h2>
            </div>
          </div>
          <div
            class="row"
          >
            <h4>
              Loading...
            </h4>
          </div>
          <div
            class="row"
          >
            <div
              class="col"
            >
              <a
                class="allEvents"
                href="/news"
              >
                Все новости
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        class="courses"
        data-testid="centers-slider"
        id="courses"
      >
        <div
          class="container"
        >
          <div
            class="row"
          >
            <div
              class="col"
            >
              <h2
                class="coursesHeading"
              >
                Симуляционные центры
              </h2>
            </div>
          </div>
          <div
            class="row"
          >
            <div
              class="col"
            >
              <div
                class="sliderWrapper"
              >
                <div>
                  <div
                    class="slick-slider slick-initialized"
                  >
                    <div
                      class="slick-list"
                      style="height: 0px;"
                    >
                      <div
                        class="slick-track"
                        style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="row"
          >
            <div
              class="col"
            >
              <a
                class="allEvents"
                href="/centers"
              >
                Смотреть все центры
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        class="news"
        data-testid="conf-and-seminars"
        id="news"
      >
        <div
          class="container"
        >
          <div
            class="row"
          >
            <div
              class="col"
            >
              <h2
                class="newsHeading"
              >
                Конференции и семинары
              </h2>
            </div>
          </div>
          <div
            class="row"
          >
            <h4>
              Loading...
            </h4>
          </div>
          <div
            class="row"
          >
            <div
              class="col"
            >
              <a
                class="allEvents"
                href="/conferencies"
              >
                Смотреть все события
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sources"
        data-testid="sources"
        id="sources"
      >
        <div
          class="container"
        >
          <div
            class="row"
          >
            <div
              class="col"
            >
              <h2
                class="sourcesHeading"
              >
                Обучающие материалы
              </h2>
            </div>
          </div>
          <div
            class="row"
          >
            <div
              class="sources"
            >
              <div
                class="me-auto mb-5 sourcesItem"
              >
                <div
                  class="sourcesIcon"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-book "
                    data-icon="book"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 448 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M96 0C43 0 0 43 0 96V416c0 53 43 96 96 96H384h32c17.7 0 32-14.3 32-32s-14.3-32-32-32V384c17.7 0 32-14.3 32-32V32c0-17.7-14.3-32-32-32H384 96zm0 384H352v64H96c-17.7 0-32-14.3-32-32s14.3-32 32-32zm32-240c0-8.8 7.2-16 16-16H336c8.8 0 16 7.2 16 16s-7.2 16-16 16H144c-8.8 0-16-7.2-16-16zm16 48H336c8.8 0 16 7.2 16 16s-7.2 16-16 16H144c-8.8 0-16-7.2-16-16s7.2-16 16-16z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
                <h3
                  class="sourcesTitle"
                >
                  Литература
                </h3>
                <p
                  class="sourcesText"
                >
                  Методические пособия, книги, презентации
                </p>
                <a
                  class="link"
                  href="/login"
                >
                  Подробнее...
                </a>
              </div>
              <div
                class="me-auto mb-5 sourcesItem"
              >
                <div
                  class="sourcesIcon"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-video "
                    data-icon="video"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 576 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 128C0 92.7 28.7 64 64 64H320c35.3 0 64 28.7 64 64V384c0 35.3-28.7 64-64 64H64c-35.3 0-64-28.7-64-64V128zM559.1 99.8c10.4 5.6 16.9 16.4 16.9 28.2V384c0 11.8-6.5 22.6-16.9 28.2s-23 5-32.9-1.6l-96-64L416 337.1V320 192 174.9l14.2-9.5 96-64c9.8-6.5 22.4-7.2 32.9-1.6z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
                <h3
                  class="sourcesTitle"
                >
                  Видеоматериалы
                </h3>
                <p
                  class="sourcesText"
                >
                  Обучающие видеоролики, конференции, лекции
                </p>
                <a
                  class="link"
                  href="/login"
                >
                  Подробнее...
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
